pkg/runtime/serializer/protobuf/protobuf_test.go

```diff
@@ -0,0 +1,187 @@
+// +build proto
+
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package protobuf_test
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/runtime/serializer/protobuf"
+)
+
+type testObject struct {
+	gvk *unversioned.GroupVersionKind
+}
+
+func (d *testObject) GetObjectKind() unversioned.ObjectKind                 { return d }
+func (d *testObject) SetGroupVersionKind(gvk *unversioned.GroupVersionKind) { d.gvk = gvk }
+func (d *testObject) GroupVersionKind() *unversioned.GroupVersionKind       { return d.gvk }
+
+type testMarshalable struct {
+	testObject
+	data []byte
+	err  error
+}
+
+func (d *testMarshalable) Marshal() ([]byte, error) {
+	return d.data, d.err
+}
+
+type testBufferedMarshalable struct {
+	testObject
+	data []byte
+	err  error
+}
+
+func (d *testBufferedMarshalable) Marshal() ([]byte, error) {
+	return nil, fmt.Errorf("not invokable")
+}
+
+func (d *testBufferedMarshalable) MarshalTo(data []byte) (int, error) {
+	copy(data, d.data)
+	return len(d.data), d.err
+}
+
+func (d *testBufferedMarshalable) Size() int {
+	return len(d.data)
+}
+
+func TestRecognize(t *testing.T) {
+	s := protobuf.NewSerializer(nil, nil, "application/protobuf")
```

Comment:
Just for testing purposes - I can use a completely arbitrary one here
instead.

On Wed, Mar 30, 2016 at 5:57 PM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/runtime/serializer/protobuf/protobuf_test.go
> <https://github.com/kubernetes/kubernetes/pull/20377#discussion_r57971034>
> :
>
> > +
> > +func (d *testBufferedMarshalable) Marshal() ([]byte, error) {
> > +	return nil, fmt.Errorf("not invokable")
> > +}
> > +
> > +func (d *testBufferedMarshalable) MarshalTo(data []byte) (int, error) {
> > +	copy(data, d.data)
> > +	return len(d.data), d.err
> > +}
> > +
> > +func (d *testBufferedMarshalable) Size() int {
> > +	return len(d.data)
> > +}
> > +
> > +func TestRecognize(t *testing.T) {
> > +	s := protobuf.NewSerializer(nil, nil, "application/protobuf")
>
> Please use a constant: defaultProtobufContentType instead
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20377/files/d7a1e247a94bf7eef7f493d96d603644e664d365#r57971034>
>